### PR TITLE
fix(angular,vue): accessible error message with aria-describedBy

### LINKED ISSUE #1737: Accessibility: Authenticator error messages should use `aria-errormessage`
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React, Angular, Vue

### Which UI component?

Authenticator

### How is your app built?

N/A

### Please describe your bug.

When Authenticator fields are invalid and have `aria-invalid` equal to true, they should also have the `aria-errormessage` attribute which points to the error message text. This allows screenreaders to announce an error message on invalid field entry.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage

### What's the expected behaviour?

`aria-errormessage` on invalid fields should point to DOM element which contain the corresponding error message. 

### Help us reproduce the bug!

1. Go to https://ui.docs.amplify.aws/components/authenticator#sign-up-attributes,  Sign Up Attributes section
2. Enter two passwords which don't match to trigger error
3. Inspect input element and see that no `aria-errormessage` is there.

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes
This follows up #1945 to add `aria-describedBy` to angular and vue.

#### Issue #, if available
Closes #1737

#### Description of how you validated changes
New e2e test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jacoblogan — APPROVED** [review]
LGTM

File: packages/vue/src/components/primitives/base-form-field.vue [hunk 1]
```diff
@@ -23,8 +24,14 @@ const { validationErrors } = toRefs(useAuthShared());
 
 const { type } = formField.value;
 
-const errors = computed(() => getErrors(validationErrors.value[name.value]));
 const isPasswordField = type === 'password';
+
+const errors = computed(() => getErrors(validationErrors.value[name.value]));
+const hasError = computed(() => errors.value?.length > 0);
+const errorId = computed(() => nanoid(12));
```
Comment (ErikCH) [inline packages/vue/src/components/primitives/base-form-field.vue:31]: Does the nanoid ever change? Else you don't need this in a computed...

Comment (wlee221) [reply packages/vue/src/components/primitives/base-form-field.vue:31]: Ooh, good catch!

Comment (wlee221) [reply packages/vue/src/components/primitives/base-form-field.vue]: c06c77404ca847c4072fff0ea1577d986808d4d1

**Review by ErikCH — APPROVED** [review]
Looks good after you fix the nanoid issue, or revert the vue change to computed.

**Pull request merged**